lib/galaxy/tools/__init__.py

```diff
@@ -1093,13 +1094,37 @@ def __ensure_help(self):
             if self.__help is HELP_UNINITIALIZED:
                 self.__inititalize_help()
 
+    def translate(self, msg):
+        if not self.app.config.enable_beta_tool_translation:
+            return msg
+
+        locales_dir = os.path.join(os.path.dirname(os.path.abspath(self.config_file)), "locales")
+        if not os.path.exists(locales_dir):
+            return msg
+
+        import gettext
+        gettext.bindtextdomain('messages', locales_dir)
+        gettext.textdomain('messages')
+
+        if msg is not None:
+            msglist = msg.split('\n')
+            newmsg = ''
+            for i in range(len(msglist)):
+                if len(msglist[i]) != 0:
+                    msglist[i] = gettext.gettext(msglist[i])
+                newmsg = newmsg + msglist[i] + '\n'
```

Comment:
The if statement for zero length seems pointless as ``gettext.gettext('')`` appears to return ``''``, so could this be much simpler, e.g.:

```python
if msg is not None:
    msg = '\n'.join(gettext.gettext(line) for line in msg.split('\n'))
```